core/src/test/java/org/elasticsearch/common/PriorityTests.java

```diff
@@ -82,11 +81,11 @@ public void testCompareTo() {
         for (Priority p : Priority.values()) {
             assertEquals(0, p.compareTo(p));
         }
-        List<Priority> shuffeledAndSorted = new ArrayList<>(Priority.values());
+        List<Priority> shuffeledAndSorted = Arrays.asList(Priority.values());
         Collections.shuffle(shuffeledAndSorted, random());
         Collections.sort(shuffeledAndSorted);
         for (List<Priority> priorities : Arrays.asList(shuffeledAndSorted,
-            Priority.values())) { // #values() guarantees order!
+            Arrays.asList(Priority.values()))) { // #values() guarantees order!
```

Comment:
@nik9000 Sorry, I think that is a mistake. The case of `Priority` is a perfect example. The value `IMMEDIATE` was added after the initial version of `Priority`. Had `Priority` been an enum, and had we been relying on ordinals, maintenance trouble would arise.